api/types/plugin.go

```diff
@@ -596,3 +613,12 @@ func (c *PluginOAuth2AccessTokenCredentials) CheckAndSetDefaults() error {
 func (c PluginStatusV1) GetCode() PluginStatusCode {
 	return c.Code
 }
+
+// CheckAndSetDefaults checks that the required fields for the Gitlab plugin are set.
+func (c *PluginGitlabSettings) Validate() error {
```

Comment:
`CheckAndSetDefaults()`, unless we're migrating away from that convention? 🙂

Reply:
We are moving away from CheckAndSetDefaults